Result of `nixpkgs-review pr 281163` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>zrythm</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>lite-xl</li>
    <li>micromamba</li>
    <li>reproc</li>
    <li>sonic-pi</li>
  </ul>
</details>
The `zrythm` failure looks unrelated (?):

```
error: builder for '/nix/store/18ny1f7x0vddjmprjr1h1pcmcw9gvj2d-zrythm-1.0.0-beta.4.9.1.drv' failed with exit code 1;
       last 10 log lines:
       > In file included from ../ext/nanovg/fontstash.h:263:
       > ../ext/nanovg/stb_truetype.h: In function 'stbtt_GetGlyphSDF':
       > ../ext/nanovg/stb_truetype.h:4675:54: warning: 'res[2]' may be used uninitialized [-Wmaybe-uninitialized]
       >  4675 |                         px = it*it*x0 + 2*t*it*x1 + t*t*x2;
       >       |                                                     ~^~
       > ../ext/nanovg/stb_truetype.h:4630:28: note: 'res[2]' was declared here
       >  4630 |                      float res[3],px,py,t,it;
       >       |                            ^~~
       > [97/574] Generating data/appdata-xml-in-tmp with a custom command
       > ninja: build stopped: subcommand failed.
```

--- a/pkgs/development/libraries/reproc/default.nix
+++ b/pkgs/development/libraries/reproc/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchFromGitHub, cmake
+{ stdenv, lib, fetchFromGitHub, cmake, fetchpatch
 }:
 
 stdenv.mkDerivation rec {
@@ -11,6 +11,19 @@ stdenv.mkDerivation rec {
     rev = "v${version}";
     sha256 = "sha256-LWzBeKhE7cSiZsK8xWzoTdrOcPiU/zEkmi40WiFytic=";
   };
+
+  patches = [
+    (fetchpatch{
+      name = "reproc-gcc-13-2.patch";
+      url = "https://github.com/DaanDeMeyer/reproc/commit/0b23d88894ccedde04537fa23ea55cb2f8365342.patch";
+      sha256 = "sha256-QyC0UcKAWCKSvSvyZTLI2eF/TuuqbGGH6cOQrS2DiCE=";
+    })
+    (fetchpatch{
+      name = "reproc-gcc-13-1.patch";
+      url = "https://github.com/DaanDeMeyer/reproc/commit/9f399675b821e175f85ac3ee6e3fd2e6056573eb.patch";
+      sha256 = "sha256-h/gnDFPWPpUFkys10YXjjEPibgRT1atHSVwbO0kId+U=";
+    })
+  ];
 
   nativeBuildInputs = [ cmake ];
 

